pywr-schema/src/nodes/core.rs

```diff
@@ -136,23 +136,176 @@ impl TryFrom<InputNodeV1> for InputNode {
     }
 }
 
+/// Cost and flow metric for soft node's constraints
+#[derive(serde::Deserialize, serde::Serialize, Clone, Default, Debug, JsonSchema, PywrVisitAll)]
+pub struct SoftConstraint {
+    pub cost: Option<Metric>,
+    pub flow: Option<Metric>,
+}
+
+#[doc = svgbobdoc::transform!(
+/// A node with cost, and min and max flow constraints. The node `L`, when connected to an upstream
+/// node `U` and downstream node `D`, will look like this on the model schematic:
+///
+/// ```svgbob
+///
+///          U         L         D
+///    - - ->*-------->*--------->*- - -
+/// ```
+///
+/// # Soft constraints
+/// This node allows setting optional maximum and minimum soft constraints which means the node's `min_flow`
```

Comment:
I think this paragraph is no longer true with how you've refactored it. The node should always stay within the min and max flow bounds now?